isolate.sh

```diff
@@ -526,6 +539,89 @@ function cleanup_webhook() {
 
 }
 
+function wait_for_certmanager() {
+    local namespace=$1
+    title " Wait for Cert Manager pods to come ready in namespace $namespace "
+    msg "-----------------------------------------------------------------------"
+    
+    #check cert manager operator pod
+    local check_retries=10
+    local name=$(${OC} get pod -n $namespace | (grep ibm-cert-manager-operator || echo "fail") | awk '{print $1}')
+    while [[ $name == "fail" ]] && [ $check_retries -gt 0 ]; do
+        sleep 15
+        name=$(${OC} get pod -n $namespace | (grep ibm-cert-manager-operator || echo "fail") | awk '{print $1}')
+        debug1 "cert manager operator pod present: $name"
+        check_retries=$(( check_retries - 1 ))
+        if [ $check_retries -eq 0 ]; then
+            error "Cert manager operator pod not found in namespace $namespace"
+        fi
+    done
```

Comment:
The name var is not set yet so if we do not wait for that, the condition gets sent through with an empty name variable. We have to wait for the cert manager pods to start before we can grep them

Reply:
I agree it seems unnecessary but it's either wait for a successful grep on the pod name or set an arbitrary sleep time to wait for cs operator to create the new cert manager deployment in control namespace and hope the following grep to set the name var works

Reply:
why does name have to be exact? The wait_for_condition does a grep anyways, so wouldn't hard coding the variable to `ibm-cert-manager-operator` be good enough?

Reply:
I think it is about `set -o errexit`, cmd `grep` will return non-zero status code if its value is not found.

So when `grep` is failed to find cert-manager-operator pod, it will abort the script immediately.

In order to solve the issue, we could make sure that `grep` is always returning zero status code. Here is an example
https://github.com/IBM/ibm-common-service-operator/blob/56743003b3ae99ec076b70ef2ce2db8947a347d2/isolate.sh#LL471C5-L471C178

Reply:
Yea, and that's being done in `check_odlm_env`, so may as well follow that and remove the first loop that checks for existence, and just go straight to checking condition.

Also, `oc` is not using the variable form

Reply:
check_odlm_env is looking for the deployment, I guess we could probably check the deployments for cert manager as well instead of pods. That would circumvent the name var unset problem.